File: src/java/Schematron.Insight/src/main/java/jp/gr/java_conf/schkit/Result.java
package jp.gr.java_conf.schkit;

import java.nio.file.Path;

import javax.management.relation.Role;

import jp.gr.java_conf.schkit.enums.ResultStatus;

/**
 * Validation Result Class
 *
 */
public class Result {
	// private variable
	private Path xmlFile;
	private Path schFile;
	private String xmlName;
	private String schName;
	private ResultStatus status;
	private String location;
	private int line;
	private int pos;
	private String message;
	private String test;
	private Role role;
	// getter setter

	/**
	 * @return validation xml file
	 */
	public Path getXmlFile() {
		return xmlFile;
	}

	/**
	 * @param validation xml file
	 */
	public void setXmlFile(Path xmlfile) {
		if (this.xmlFile != xmlfile) {
			this.xmlFile = xmlfile;
			this.xmlName = xmlfile.getFileName().toString();
		}
	}

	/**
	 * @return schema file
	 */
	public Path getSchFile() {
		return schFile;
	}

	/**
	 * @param schema file
	 */
	public void setSchFile(Path schfile) {
		if (this.schFile != schfile) {
			this.schFile = schfile;
			this.schName = schfile.getFileName().toString();
		}
	}


	/**
	 * @return validation xml file name
	 */
	public String getXmlName() {
		return xmlName;
	}

	/**
	 * @return validation xml file name
	 */
	public String getSchName() {
		return schName;
	}

	/**
	 * @return validation result status
	 */
	public ResultStatus getStatus() {
		return status;
	}
	/**
	 * @param validation result status
	 */
	public void setStatus(ResultStatus status) {
		this.status = status;
	}

	/**
	 * @return validation result status display name
	 */
	public String getStatusName()
	{
		return status.getName();
	}
	/**
	 * @param validation result status display name
	 */
	public void setStatusName(String name)
	{
		status = ResultStatus.fromName(name);
	}

	/**
	 * @return detected xpath location
	 */
	public String getLocation() {
		return location;
	}

	/**
	 * @param detected xpath location
	 */
	public void setLocation(String location) {
		this.location = location;
	}

	/**
	 * @return detected text line number
	 */
	public int getLine() {
		return line;
	}

	/**
	 * @param detected text line number
	 */
	public void setLine(int line) {
		this.line = line;
	}

	/**
	 * @return detected text position
	 */
	public int getPos() {
		return pos;
	}

	/**
	 * @param detected text position
	 */
	public void setPos(int pos) {
		this.pos = pos;
	}

	/**
	 * @return result report text
	 */
	public String getMessage() {
		return message;
	}

	/**
	 * @param result report text
	 */
	public void setMessage(String message) {
		this.message = message;
	}

	/**
	 * @return schematron test phase name
	 */
	public String getTest() {
		return test;
	}

	/**
	 * @param schematron test phase name
	 */
	public void setTest(String test) {
		this.test = test;
	}

	/**
	 * @return schematron role infomation
	 */
	public Role getRole() {
		return role;
	}

	/**
	 * @param schematron role infomation
	 */
	public void setRole(Role role) {
		this.role = role;
	}
}

File: src/cs/Schematron.Insight/Validation/Result.cs
using Schematron.Insight.Utilities;
using System;
using System.IO;
using System.Runtime.Serialization;
using System.Xml;
using System.Xml.Serialization;

namespace Schematron.Insight.Validation
{
    [Serializable()]
    [DataContract]
    public class Result
    {
        private FileInfo _xmlfile;
        private FileInfo _schfile;
        [XmlIgnore]
        public FileInfo XmlFile
        {
            get { return _xmlfile; }
            set
            {
                if (_xmlfile != value)
                {
                    _xmlfile = value;
                    XmlName = value.Name;
                }
            }
        }
        [XmlIgnore]
        public FileInfo SchFile
        {
            get { return _schfile; }
            set
            {
                if (_schfile != value)
                {
                    _schfile = value;
                    SchName = value.Name;
                }
            }
        }

        [XmlElement(ElementName = "XmlFile", Order = 1)]
        [DataMember(Name = "XmlFile", Order = 1)]
        public string XmlName { get; set; } = "";
        [XmlElement(ElementName = "SchemaFile", Order = 2)]
        [DataMember(Name = "SchemaFile", Order = 2)]
        public string SchName { get; set; } = "";
        [XmlIgnore]
        [IgnoreDataMember]
        public ResultStatus Status { get; set; } = ResultStatus.None;
        [XmlElement(ElementName = "Status", Order = 3)]
        [DataMember(Name = "Status", Order = 3)]
        public string StatusName
        {
            get
            {
                return Status.DisplayName();
            }
            set
            {
                Status = EnumerationTypeHelper.GetValueFromDisplayName<ResultStatus>(value);
            }
        }
        [XmlElement(ElementName = "Location", Order = 4)]
        [DataMember(Name = "Location", Order = 4)]
        public string Location { get; set; } = "";
        [XmlElement(ElementName = "Line", Order = 5)]
        [DataMember(Name = "Line", Order = 5)]
        public int Line { get; set; } = -1;
        [XmlElement(ElementName = "Pos", Order = 6)]
        [DataMember(Name = "Pos", Order = 6)]
        public int Pos { get; set; } = -1;

        [XmlElement(ElementName = "Message", Order = 7)]
        [DataMember(Name = "Message", Order = 7)]
        public string Message { get; set; } = "";
        [XmlElement(ElementName = "Test", Order = 8)]
        [DataMember(Name = "Test", Order = 8)]
        public string Test { get; set; } = "";
        [XmlElement(ElementName = "Role", Order = 9)]
        [DataMember(Name = "Role", Order = 9)]
        public Role Role { get; set; } = new Role();
    }

}
